### PR TITLE
[feature-wip] support avro format in routine load and stream load

### LINKED ISSUE #7650: [Proposal] Support avro format load
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

In JingDong, many users use routine load data with avro format, we will be prepared to contribute it to the doris community.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Henry2SS**
# Proposed changes

Issue Number: close #7650 

## Problem Summary:

Support avro format in routine load and stream load.
There are still several places to be improved.

And tested locally for more than a week, loading is OK and no coredump, no mem leak.

Still in progress:
1. support routine load property named `avro_schema_path` to sepcified a schema file, not from a config
2. support avro::array type to doris::array
3. support broker load in the future 

## Checklist(Required)

1. Does it affect the original behavior: (No)
4. Has unit tests been added: (No)
5. Has document been added or modified: (No)
6. Does it need to update dependencies: (Yes)
7. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/avro_scanner.cpp [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+    _close();
+}
+
+Status AvroReader::init() {
+    bool exist = FileUtils::check_exist(config::avro_schema_file_path);
+    if (!exist) {
+        return Status::InternalError("there is no avro schema file at " +
+                                     config::avro_schema_file_path +
+                                     ". Please put an schema file in json format.");
+    } else {
```
Comment (yinzhijian) [inline be/src/exec/avro_scanner.cpp:198]: The else statement does not appear to be required.
```suggestion
    } 
```

File: be/src/exec/avro_scanner.cpp [hunk 2]
```diff
@@ -0,0 +1,525 @@
...
+    case avro::AVRO_ENUM:
+        [[fallthrough]];
+    case avro::AVRO_FIXED:
+        [[fallthrough]];
+    case avro::AVRO_MAP:
+        [[fallthrough]];
+    case avro::AVRO_RECORD:
+        [[fallthrough]];
+    default: {
+        throw avro::Exception("Not support " + root_node->type() + std::string("type yet."));
```
Comment (yinzhijian) [inline be/src/exec/avro_scanner.cpp:371]:
```suggestion
        throw avro::Exception("Not support " + root_node->type() + std::string(" type yet."));
```

File: be/src/exec/avro_scanner.cpp [hunk 3]
```diff
@@ -0,0 +1,525 @@
...
+    } break;
+    }
+    throw avro::Exception("Type " + slot_desc->type().type +
+                          std::string("is not compatible with Avro ") +
```
Comment (yinzhijian) [inline be/src/exec/avro_scanner.cpp:375]:
```suggestion
                          std::string(" is not compatible with Avro ") +
```

File: be/src/common/config.h [hunk 4]
```diff
@@ -763,6 +763,9 @@ CONF_Int32(quick_compaction_batch_size, "10");
 // do compaction min rowsets
 CONF_Int32(quick_compaction_min_rowsets, "10");
 
+// Avro schema file path, set default to "${DORIS_HOME}/conf/avro_schema.json"
+CONF_String(avro_schema_file_path, "${DORIS_HOME}/conf/avro_schema.json");
```
Comment (morningman) [inline be/src/common/config.h:767]: So that each Doris cluster only support one avro schema?

**Review by morningman — COMMENTED** [review]
The old query engine with `tuple` and `RowBatch` interface will be deprecated in the future.
So you'd better implement this feature in vec engine, with `block` interface.

File: be/src/common/config.h (hunk 4 above)
Comment (Henry2SS) [reply be/src/common/config.h:767]: should have a property of routine load named `avro_schema_name`, and the schema should be put on every BE.

File: be/src/exec/avro_scanner.cpp (hunk 3 above)
Comment (Henry2SS) [reply be/src/exec/avro_scanner.cpp]: Thx for your advice~ Fixed all the comments above.

File: be/src/common/config.h (hunk 4 above)
Comment (Henry2SS) [reply be/src/common/config.h]: > The old query engine with `tuple` and `RowBatch` interface will be deprecated in the future. So you'd better implement this feature in vec engine, with `block` interface.

Hi, support Vectorized `VAvroScanner` now.

**Pull request closed without merging**